src/syntax/ast.ts

```diff
@@ -542,6 +542,7 @@ export interface SelectFromStatement extends PGNode {
     limit?: LimitStatement | nil;
     orderBy?: OrderByStatement[] | nil;
     distinct?: 'all' | 'distinct' | Expr[] | nil;
+    for?: ForStatement;
```

Comment:
Why not just:

```typescript
for?: 'update' | 'no key update' | 'share' | 'key share'
```

?

Agreed, we'd be loosing the location of those keywords, but it really is not of great value (given that it can be infered).

And moreover, i would prefer spaces instead of underscores, to be coherent with the rest.

Reply:
I used a statement because the full syntax is:

[ FOR { UPDATE | NO KEY UPDATE | SHARE | KEY SHARE } [ OF table_name [, ...] ] [ NOWAIT | SKIP LOCKED ] [...] ]

I thought I start with the simple version first though... Any advice welcome!

Reply:
Okay, fair enough 👍